File: Python/perfect-number.py
class Solution:
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num <= 1:
            return False

        res = 1
        i = 2
        s = num ** 0.5

        while i <= s:
            if num % i == 0:
                res += (i + num // i)
            i += 1

        return res == num

File: Swift/perfect-number.swift
class Solution {
    func checkPerfectNumber(_ num: Int) -> Bool {
        guard num > 1 else { return false }

        var sum = 1
        var i = 2
        let s = Int(sqrt(Double(num)))

        while i <= s {
            if num % i == 0 {
                sum += (i + num / i)
            }
            i += 1
        }

        return sum == num
    }
}
